fleet/error.go

```diff
@@ -8,6 +10,18 @@ var (
 	maskAny = errgo.MaskFunc(errgo.Any)
 )
 
+func maskAnyf(err error, f string, v ...interface{}) error {
```

Comment:
The user can. The effect would simply not be the same. This function makes sure the error has the given cause and the error message is as given. I found the error message handling very weird, so I extended it.